pkgs/os-specific/linux/kernel-headers/default.nix

```diff
@@ -1,26 +1,35 @@
 { stdenvNoCC, lib, buildPackages, fetchurl, perl, elf-header }:
 
 let
-  makeLinuxHeaders = { src, version, patches ? [] }: stdenvNoCC.mkDerivation {
+  makeLinuxHeaders = { arch, src, version, patches ? [] }: stdenvNoCC.mkDerivation {
     inherit src;
 
+    inherit arch;
     pname = "linux-headers";
     inherit version;
 
-    ARCH = stdenvNoCC.hostPlatform.linuxArch;
+    ARCH = arch;
 
     # It may look odd that we use `stdenvNoCC`, and yet explicit depend on a cc.
     # We do this so we have a build->build, not build->host, C compiler.
     depsBuildBuild = [ buildPackages.stdenv.cc ];
     # `elf-header` is null when libc provides `elf.h`.
-    nativeBuildInputs = [ perl elf-header ];
+    nativeBuildInputs = [ perl elf-header ]
+    ++ lib.optional stdenvNoCC.hostPlatform.isAndroid ( with buildPackages.buildPackages; [
+      bison
+      flex
+      python
+      rsync
+    ]);
 
     extraIncludeDirs = lib.optional stdenvNoCC.hostPlatform.isPowerPC ["ppc"];
 
     inherit patches;
 
     hardeningDisable = lib.optional stdenvNoCC.buildPlatform.isDarwin "format";
 
+    sourceRoot = lib.optionalString stdenvNoCC.hostPlatform.isAndroid ".";
```

Comment:
What is this for? I would have thought this is the default?

Reply:
The kernel we get from the repo is not in a top level directory. So when we unpack we get all the directories that would normally be under that top level directory instead. Thus we need to set sourceRoot to the current directory.

Reply:
Ok